server/src/main/java/org/elasticsearch/cluster/routing/ShardMovementWriteLoadSimulator.java

```diff
@@ -30,11 +31,14 @@ public class ShardMovementWriteLoadSimulator {
     private final Map<String, NodeUsageStatsForThreadPools> originalNodeUsageStatsForThreadPools;
     private final ObjectDoubleMap<String> simulatedNodeWriteLoadDeltas;
     private final Map<ShardId, Double> writeLoadsPerShard;
+    // A map of nodeId to a boolean indicate whether it has seen a shard moving away from it
+    private final Map<String, Boolean> hasSeenMovedAwayShardMap;
 
     public ShardMovementWriteLoadSimulator(RoutingAllocation routingAllocation) {
         this.originalNodeUsageStatsForThreadPools = routingAllocation.clusterInfo().getNodeUsageStatsForThreadPools();
         this.writeLoadsPerShard = routingAllocation.clusterInfo().getShardWriteLoads();
         this.simulatedNodeWriteLoadDeltas = new ObjectDoubleHashMap<>();
+        this.hasSeenMovedAwayShardMap = new HashMap<>();
```

Comment:
Can this be a hash-set?

Reply:
Yes. Could not think straight on a Friday afternoon .. 
pushed b36753e6f3f9390adeb2f7df2704ea9d4aa5a09d